dev/com.ibm.ws.security.acme/src/com/ibm/ws/security/acme/internal/AcmeHistory.java

```diff
@@ -236,4 +237,28 @@ public void writeAcmeFileLine(File file, String date, String serial, String dire
 		}
 		return entries;
 	}
+	
+	public String getDirectoryURI(String serial) {
+		if (!acmeFile.exists()) {
+			return null;
+		}
+		try {
+		    BufferedReader br = new BufferedReader(new FileReader(acmeFile));
```

Comment:
We repeat the tokenizing of this file in several methods. Can't we just have one method that reads the files, creates some simple pojo that contains the data, and then call that from all of these methods? I think that would make it much cleaner and reduce the number of places that have the same sort of code that we need to maintain.